cylc/uiserver/logging_util.py

```diff
@@ -0,0 +1,75 @@
+# Copyright (C) NIWA & British Crown (Met Office) & Contributors.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from contextlib import suppress
+from glob import glob
+import logging
+import os
+from pathlib import Path
+
+from cylc.uiserver.app import USER_CONF_ROOT
+
+
+class RotatingUISFileHandler(logging.handlers.RotatingFileHandler):
+    """Rotate logs on ui-server restart"""
+
+    LOG_NAME_EXTENSION = "-uiserver.log"
+
+    def __init__(self):
+        self.file_path = Path(USER_CONF_ROOT / "log").expanduser()
+
+    def on_start(self):
+        """Set up logging"""
+        self.file_path.mkdir(parents=True, exist_ok=True)
+        self.delete_symlink()
+        self.rename_logs()
+        self.setup_new_log()
+        self.update_log_archive()
+
+    def update_log_archive(self):
+        """Ensure log archive retains only 5 logs"""
+        log_files = sorted(
+            glob(os.path.join(
+                self.file_path, f"*{self.LOG_NAME_EXTENSION}")
+            ), reverse=True)
+        while len(log_files) > 5:
+            os.unlink(log_files.pop(0))
+
+    def rename_logs(self):
+        """Increment the log number by one for each log"""
+        # increments each log number, done in descending order to avoid
+        # filename conflicts
+        log_files = sorted(glob(os.path.join(
+            self.file_path, f"*{self.LOG_NAME_EXTENSION}")), reverse=True)
+        for file in log_files:
+            log_num = int(Path(file).name.partition('-')[0]) + 1
```

Comment:
You might want to consider using a try/except loop around this, or making the glob a bit more careful (`[0-9]*{self.LOG_NAME_EXTENSION}`).

Otherwise I can break this thus:

touch ~/.cylc/uiserver/log/iduhafiuhbjvfaiehr-uiserver.log

Reply:
Good spot, unlikely but worth fixing.